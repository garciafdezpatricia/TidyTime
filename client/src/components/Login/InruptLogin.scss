@use "../../../styles/variables" as *;

.inrupt-container {
    width: $view-width;
    height: 100%;
    padding: $view-padding;
    box-sizing: border-box;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: $hover-green;

    .inrupt-login {
        display: flex;
        flex-direction: column;
        width: 28rem;
        height: 32rem;
        border: .1rem solid $white;
        padding: 1rem;
        background-color: $white;
        box-shadow: 0px 13px 34px 15px rgba(0,0,0,0.27);
        border-radius: .8rem;
        box-sizing: border-box;

        .inrupt-login-header {
            width: 100%;

            .inrupt-logo {
                width: 100%;
                height: 8rem;
                background-image: url("../../../public/inrupt-logo.svg");
                background-repeat: no-repeat;
                background-size: contain;
                background-position: center;
            }
        }

        .inrupt-login-body {
            display: flex;
            flex-direction: column;
            height: calc(100% - 16rem);
            align-items: center;
            gap: 1.5rem;
            text-align: center;
            h2 {
                color: $primary-green;
            }

            p {
                width: 85%;
                text-align: center;
                line-height: 1.5rem;
                font-size: .9rem;
            }

            button {
                width: 80%;
                display: flex;
                align-items: center;
                justify-content: center;
                background-color: $medium-grey;
                color: $white;
                outline: none;
                border: none;
                font-size: 1rem;
                padding: .3rem 1rem;
                &:hover {
                    background-color: $primary-green;
                }
            }
        }

        .inrupt-login-footer {
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            width: 100%;
            gap: .5rem;
            height: 50%;
            
            p {
                width: 85%;
                text-align: center;
                line-height: 1.5rem;
                font-size: .9rem;
            }
        }
    }
}
