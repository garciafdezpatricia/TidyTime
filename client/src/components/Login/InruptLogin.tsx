import { useInruptHandler } from "@/pages/api/inrupt";
import { useSessionContext } from "../Context/SolidContext";
export default function Inrupt() {

    const { loginInrupt } = useInruptHandler();
    const { solidSession } = useSessionContext();

    return (
        <section className="inrupt-container">
        {
            !solidSession?.info.isLoggedIn && 
            <article className="inrupt-login">
                <section className="inrupt-login-header">
                    <div className="inrupt-logo"></div>
                </section>
                <section className="inrupt-login-body">
                    <h2>Welcome to TidyTime!</h2>
                    <p>TidyTime works with Solid PODs, ensuring your data is only controlled by you. Log in to your Inrupt Solid POD and start using the app!</p>
                    <button onClick={loginInrupt}>LOGIN</button>
                </section>
                <section className="inrupt-login-footer">
                    <p>Learn more about <a href="https://www.inrupt.com/products/enterprise-solid-server">Inrupt&apos;s ESS</a>
                    <br />Learn more about the <a href="https://solidproject.org/">Solid Project</a>
                    </p>
                </section>
            </article>
        }
        </section>
    )
}