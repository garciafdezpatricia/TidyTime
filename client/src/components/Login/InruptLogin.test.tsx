import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import InruptLogin from '@/src/components/Login/InruptLogin';

describe('InruptLogin', () => {
    it('renders without crashing', () => {
      const { container } = render(<InruptLogin/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<InruptLogin/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });