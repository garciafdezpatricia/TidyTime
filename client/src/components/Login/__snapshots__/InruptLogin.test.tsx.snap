// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InruptLogin matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="inrupt-container"
  >
    <article
      class="inrupt-login"
      data-testid="inrupt-login-article"
    >
      <section
        class="inrupt-login-header"
      >
        <div
          class="inrupt-logo"
        />
      </section>
      <section
        class="inrupt-login-body"
      >
        <h2>
          loginPage.title
        </h2>
        <p>
          loginPage.desc
        </p>
        <button
          data-testid="inrupt-login-button"
        >
          loginPage.loginButton
        </button>
      </section>
      <section
        class="inrupt-login-footer"
      >
        <p>
          loginPage.learnMore 
          <a
            href="https://www.inrupt.com/products/enterprise-solid-server"
          >
            loginPage.inruptESS
          </a>
          <br />
          loginPage.learnMore
          <a
            href="https://solidproject.org/"
          >
            loginPage.solidProject
          </a>
        </p>
      </section>
    </article>
  </section>
</DocumentFragment>
`;
